android/java/org/chromium/chrome/browser/settings/BraveSyncScreensPreference.java

```diff
@@ -69,1155 +69,901 @@
+            try {
+                int code = GoogleApiAvailability.getInstance().isGooglePlayServicesAvailable(
+                        getActivity().getApplicationContext());
+                if (code != ConnectionResult.SUCCESS) {
+                    Dialog dlg = GoogleApiAvailability.getInstance().getErrorDialog(
+                            getActivity(), code, RC_HANDLE_GMS);
+                    if (null != dlg) {
+                        dlg.show();
+                    }
+                }
+            } catch (ActivityNotFoundException e) {
+                Log.e(TAG, "Unable to start camera source.", e);
+                mCameraSource.release();
+                mCameraSource = null;
 
-  private void cancelTimeoutTimer() {
-      mTimeoutTimer.cancel();
-  }
+                return;
+            }
+            try {
+                mCameraSourcePreview.start(mCameraSource);
+            } catch (IOException e) {
+                Log.e(TAG, "Unable to start camera source.", e);
+                mCameraSource.release();
+                mCameraSource = null;
+            }
+        }
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        try {
+            if (null != mCameraSourcePreview
+                    && View.GONE != mScrollViewSyncChainCode.getVisibility()) {
+                startCameraSource();
+            }
+        } catch (SecurityException se) {
+            Log.e(TAG, "Do not have permission to start the camera", se);
+        } catch (RuntimeException e) {
+            Log.e(TAG, "Could not start camera source.", e);
+        }
+    }
 
-  private void showAddDeviceNameDialog(boolean createNewChain) {
-      LayoutInflater inflater = (LayoutInflater) getActivity().getSystemService(
-              Context.LAYOUT_INFLATER_SERVICE);
-      View view = inflater.inflate(R.layout.add_sync_device_name_dialog, null);
-      final EditText input = (EditText) view.findViewById(R.id.device_name);
+    @Override
+    public void onPause() {
+        super.onPause();
+        if (mCameraSourcePreview != null) {
+            mCameraSourcePreview.stop();
+        }
+        InputMethodManager imm =
+                (InputMethodManager) getActivity().getSystemService(Context.INPUT_METHOD_SERVICE);
+        imm.hideSoftInputFromWindow(getView().getWindowToken(), 0);
+    }
+
+    @Override
+    public void onDestroy() {
+        super.onDestroy();
+        if (mCameraSourcePreview != null) {
+            mCameraSourcePreview.release();
+        }
+        if (deviceInfoObserverSet) {
+            BraveSyncDevices.get().removeDeviceInfoChangedListener(this);
+            deviceInfoObserverSet = false;
+        }
+    }
+
+    // Barcode is valid when its raw representation has length of 64
+    // and when it's possible to convert the barcode to bip39 code words
+    private boolean isBarCodeValid(String barcode) {
+        Log.i(TAG, "isBarCodeValid barcode length=" + barcode.length());
```

Comment:
let's remove unneeded logs

Reply:
addressed in 0c7a7e42652e4bf75eac1c2f69d19cb01344d64d